src/seedu/addressbook/AddressBook.java

```diff
@@ -315,34 +319,53 @@ private static boolean isValidFilePath(String filePath) {
         if (filePath == null) {
             return false;
         }
-        Path filePathToValidate;
+        return isValid(filePath);
+    }
+
+	private static boolean isValid(String filePath) {
+		Path filePathToValidate;
         try {
             filePathToValidate = Paths.get(filePath);
         } catch (InvalidPathException ipe) {
             return false;
         }
         return hasValidParentDirectory(filePathToValidate) && hasValidFileName(filePathToValidate);
-    }
+	}
 
     /**
      * Returns true if the file path has a parent directory that exists.
      */
     private static boolean hasValidParentDirectory(Path filePath) {
         Path parentDirectory = filePath.getParent();
-        return parentDirectory == null || Files.isDirectory(parentDirectory);
+        return isNull(parentDirectory) || isDirectory(parentDirectory);
```

Comment:
isNull is not necessary actually. There are many different data types that we can check for `null`. would that mean that you need to write many different overloaded `isNull` methods to support this?